src/hooks/tradingView/useChartLines.ts

```diff
@@ -0,0 +1,206 @@
+
+    const entryPrice = currentMarketPositionData.entryPrice?.current;
+    const size = currentMarketPositionData.size?.current;
+
+    const key = currentMarketPositionData.id;
+    const price = MustBigNumber(entryPrice).toNumber();
+
+    const maybePositionLine = chartLines[key]?.line;
+    const shouldShow = size && size !== 0;
+
+    if (!shouldShow) {
+      if (maybePositionLine) {
+        maybePositionLine.remove();
+        delete chartLines[key];
+        return;
+      }
+    } else {
+      const quantity = size.toString();
+
+      if (maybePositionLine) {
+        if (maybePositionLine.getQuantity() !== quantity) {
+          maybePositionLine.setQuantity(quantity);
+        }
+        if (maybePositionLine.getPrice() !== price) {
+          maybePositionLine.setPrice(price);
+        }
+      } else {
+        const positionLine = tvWidget
+          ?.chart()
+          .createPositionLine({ disableUndo: false })
+          .setText(stringGetter({ key: STRING_KEYS.ENTRY_PRICE_SHORT }))
+          .setPrice(price)
+          .setQuantity(quantity);
+
+        if (positionLine) {
+          const chartLine = { line: positionLine, chartLineType: 'position' };
+          setLineColors({ chartLine: chartLine });
+          chartLines[key] = chartLine;
+        }
+      }
+    }
+  };
+
+  const drawOrderLines = () => {
+    if (!currentMarketOrders) return;
+
+    currentMarketOrders.forEach(
+      ({
+        id,
+        type,
+        status,
+        side,
+        cancelReason,
+        remainingSize,
+        size,
+        triggerPrice,
+        price,
+        trailingPercent,
+      }) => {
+        const key = id;
+        const quantity = (remainingSize ?? size).toString();
+
+        const orderType = type.rawValue as OrderType;
+        const orderLabel = stringGetter({
+          key: ORDER_TYPE_STRINGS[orderType].orderTypeKey,
+        });
+        const orderString = trailingPercent ? `${orderLabel} ${trailingPercent}%` : orderLabel;
+
+        const shouldShow =
+          !cancelReason &&
+          (status === AbacusOrderStatus.open || status === AbacusOrderStatus.untriggered);
+
+        const maybeOrderLine = chartLines[key]?.line;
+
+        if (!shouldShow) {
+          if (maybeOrderLine) {
+            maybeOrderLine.remove();
+            delete chartLines[key];
+            return;
+          }
+        } else {
+          if (maybeOrderLine) {
+            if (maybeOrderLine.getQuantity() !== quantity) {
+              maybeOrderLine.setQuantity(quantity);
+            }
+          } else {
+            const orderLine = tvWidget
+              ?.chart()
+              .createOrderLine({ disableUndo: false })
+              .setPrice(MustBigNumber(triggerPrice ?? price).toNumber())
+              .setQuantity(quantity)
+              .setText(orderString);
+
+            if (orderLine) {
+              const chartLine = { line: orderLine, chartLineType: ORDER_SIDES[side.name] };
```

Comment:
I think `chartLine` needs to be typed
<img width="1114" alt="Screen Shot 2024-02-13 at 2 49 34 PM" src="https://github.com/dydxprotocol/v4-web/assets/13111994/b5aeffb8-de93-4a6c-93d6-87a291ecde74">

```suggestion
              const chartLine: ChartLine = { line: orderLine, chartLineType: ORDER_SIDES[side.name] };
```